docs/qppe.yaml

```diff
@@ -806,3 +825,148 @@ components:
           description: Human-readable reason why there was an error.
       required: [ code ]
       additionalProperties: false
+
+    FormLabel:
+      type: string
+      description: The displayed label for this form element
+    FormName:
+      type: string
+      pattern: ^[a-zA-Z_][a-zA-Z0-9_]*$
+      description: The key of the form data field produced by this form element
+
+    TextElement:
+      type: object
+      properties:
+        kind:
+          type: string
+          enum: [ text ]
+        label:
+          $ref: "#/components/schemas/FormLabel"
+        text:
+          type: string
+          description: Static text to display
+      required: [ kind, label, text ]
+
+    TextInputElement:
+      type: object
+      properties:
+        kind:
+          type: string
+          enum: [ input ]
+        label:
+          $ref: "#/components/schemas/FormLabel"
+        name:
+          $ref: "#/components/schemas/FormName"
+        required:
+          type: boolean
+          default: false
+      required: [ kind, label, name ]
+
+    CheckboxElement:
+      type: object
+      properties:
+        kind:
+          type: string
+          enum: [ checkbox ]
+        label:
```

Comment:
Was ist hier das label? In den Moodle Forms können Checkboxen zwei label haben: https://docs.moodle.org/dev/lib/formslib.php_Form_Definition#checkbox

Reply:
Ich habe das jetzt durch left_label und right_label ersetzt. Ich bin zwar mit den Namen nicht so ganz happy (andere LMS ordnen labels vielleicht generell nicht neben den Elementen an, oder haben eben nur eines für checkboxen), aber jetzt soll es erstmal reichen